src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -154,6 +154,16 @@ Map<String, Object> buildMapWithProcessorKeyAndOriginalValue(IngestDocument inge
         for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
             String originalKey = fieldMapEntry.getKey();
             Object targetKey = fieldMapEntry.getValue();
+
+            int nestedDotIndex = originalKey.indexOf('.');
+            if (nestedDotIndex != -1) {
+                Map<String, Object> temp = new LinkedHashMap<>();
```

Comment:
why do we need this map object? Can you please use more meaningful name for map variable

Reply:
I added a more meaningful name for the map variable.